src/server/client.rs

```diff
@@ -223,9 +223,50 @@ async fn handle_autoalloc_message(
         AutoAllocRequest::AddQueue(params) => create_queue(server_dir, state_ref, params).await,
         AutoAllocRequest::Events { descriptor } => get_event_log(state_ref, descriptor),
         AutoAllocRequest::Info { descriptor } => get_allocations(state_ref, descriptor),
+        AutoAllocRequest::RemoveQueue(name) => remove_queue(state_ref, name).await,
     }
 }
 
+async fn remove_queue(state_ref: &StateRef, id_to_delete: DescriptorId) -> ToClientMessage {
+    let futures: Vec<_> = {
+        state_ref
+            .get()
+            .get_autoalloc_state()
+            .descriptors()
+            .filter(|(id, _)| *id == id_to_delete)
+            .map(|(_, descriptor)| {
+                let handler = descriptor.descriptor.handler();
+                descriptor
+                    .all_allocations()
+                    .filter(|alloc| alloc.is_active())
+                    .map(move |alloc| {
+                        let fut = handler.remove_allocation(alloc.id.clone());
+                        let id = alloc.id.clone();
+                        async move { (fut.await, id) }
+                    })
+            })
+            .flatten()
+            .collect()
+    };
+
+    for (result, allocation_id) in futures::future::join_all(futures).await {
+        match result {
+            Ok(_) => {
+                log::info!("Allocation {} was removed", allocation_id);
+            }
+            Err(e) => {
+                log::error!("Failed to remove allocation {}: {:?}", allocation_id, e);
+            }
+        }
+    }
+    state_ref
```

Comment:
The descriptor should be deleted before `futures::future::join_all(futures).await` (if possible), otherwise new allocations can be created in the meantime.